dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -211,6 +214,21 @@ def is_match(self) -> bool:
             is_match = True
         return is_match
 
+    def check_stop_condition_is_met(self, data: DataFrame) -> bool:
+        """Return value stop_condition_is_met provided the new data being added
+        :param data: Dataframe currently being processed by categorical profiler
+        :type data: DataFrame
+        :return: boolean for stop conditions
+        """
+        if not self._stop_condition_is_met:
+            total_categories = len(self._categories) + len(data.value_counts())
```

Comment:
maybe we should update the bool after we do the calc and merge. Otherwise:
1. we calc value_counts twice.
2. it will check the actual merge length which accounts for overlap